crates/common/tedge_config/src/tedge_config_cli/tedge_config.rs

```diff
@@ -318,6 +318,88 @@ impl TEdgeTomlVersion {
     }
 }
 
+pub trait AppendRemoveItem {
+    type Item;
+
+    fn append(current_value: Option<Self::Item>, new_value: Self::Item) -> Option<Self::Item>;
+
+    fn remove(current_value: Option<Self::Item>, remove_value: Self::Item) -> Option<Self::Item>;
+}
+
+macro_rules! impl_append_remove_for_single_value {
+    ($($type:ty),*) => {
+        $(
+            impl AppendRemoveItem for $type {
+                type Item = $type;
+
+                fn append(_current_value: Option<Self::Item>, new_value: Self::Item) -> Option<Self::Item> {
+                    Some(new_value)
+                }
+
+                fn remove(current_value: Option<Self::Item>, remove_value: Self::Item) -> Option<Self::Item> {
+                    match current_value {
+                        Some(current) if current == remove_value => None,
+                        _ => current_value
+                    }
+                }
+            }
+        )*
+    }
+}
+
+impl_append_remove_for_single_value!(
```

Comment:
As suggested by @jarhodes314, it would be really good to add a [custom diagnostic](https://blog.rust-lang.org/2024/05/02/Rust-1.78.0.html#diagnostic-attributes) to the `AppendRemoveItem` telling the type of a config setting should listed here.

To have something like that:

```
error[E0277]: To be used as a tedge config type, connect_url::ConnectUrl must implement the trait `AppendRemoveItem`
   --> crates/common/tedge_config/src/tedge_config_cli/tedge_config.rs:617:14
    |
617 |         url: ConnectUrl,
    |              ^^^^^^^^^^ the trait `AppendRemoveItem` is not implemented for `connect_url::ConnectUrl`
    |
    = note: This can be done using impl_append_remove_for_single_value!
```

instead of:

```
error[E0277]: the trait bound `connect_url::ConnectUrl: AppendRemoveItem` is not satisfied
   --> crates/common/tedge_config/src/tedge_config_cli/tedge_config.rs:614:14
    |
614 |         url: ConnectUrl,
    |              ^^^^^^^^^^ the trait `AppendRemoveItem` is not implemented for `connect_url::ConnectUrl`

```